docs/content/dagster-cloud/managing-deployments/controlling-logs.mdx

```diff
@@ -0,0 +1,96 @@
+
+### Writing to your own S3 bucket
+
+If using the Kubernetes agent, you can instead forward logs to your own S3 bucket by using the [`S3ComputeLogManager`](/\_apidocs/libraries/dagster-aws#dagster_aws.s3.S3ComputeLogManager).
+
+You can configure the `S3ComputeLogManager` in your [`dagster.yaml` file](https://docs.dagster.io/dagster-cloud/deployment/agents/customizing-configuration):
+
+```yaml
+compute_logs:
+  module: dagster_aws.s3.compute_log_manager
+  class: S3ComputeLogManager
+  config:
+    show_url_only: true
+    bucket: your-compute-log-storage-bucket
+    region: your-bucket-region
+```
+
+If you are using Helm to deploy thew Kubernetes agent, you can provide the following configuration in your `values.yaml` file:
+
+```yaml
+computeLogs:
+  enabled: true
+  custom:
+    module: dagster_aws.s3.compute_log_manager
+    class: S3ComputeLogManager
+    config:
+      show_url_only: true
+      bucket: your-compute-log-storage-bucket
+      region: your-bucket-region
+```
+
+### Disabling compute log upload
+
+If your organization has its own logging solution which ingests stdout and stderr from your compute environment, you may want to disable compute log upload entirely. You can do this with the [`NoOpComputeLogManager`](/\_apidocs/internals#dagster.\_core.storage.noop_compute_log_manager.NoOpComputeLogManager).
+
+You can configure the `NoOpComputeLogManager` in your [`dagster.yaml` file](https://docs.dagster.io/dagster-cloud/deployment/agents/customizing-configuration):
+
+```yaml
+compute_logs:
+  module: dagster.core.storage.noop_compute_log_manager
+  class: NoOpComputeLogManager
+```
+
+If you are using Helm to deploy thew Kubernetes agent, the `enabled` flag lets you easily disable compute log upload:
+
+```yaml
+computeLogs:
+  enabled: false
+```
+
+### Other compute log storage options
+
+For a full list of available compute log storage options, refer to the [Compute log storage docs](/deployment/dagster-instance#compute-log-storage).
+
+---
+
+## Masking error messages
+
+By default, Dagster Cloud surfaces error messages from your code locations in the Dagster Cloud UI, including when runs fail, when sensors or schedules throw an exception, or when your code locations fail to load. You can mask these error messages in the case that their contents are sensitive.
+
+In order to mask error messages in a Dagster Cloud Deployment, you can set the environment variable `DAGSTER_REDACT_USER_CODE_ERRORS` equal to `1` using the [Dagster Cloud Environment Variables](/dagster-cloud/managing-deployments/environment-variables-and-secrets) UI:
+
+<Image
+alt="Environment variable UI showing DAGSTER_REDACT_USER_CODE_ERRORS set to 1"
+src="/images/dagster-cloud/logs-error-messages/configure-redact-env-var.png"
+width={644}
+height={407}
+/>
+
+Once set, error messages from your code locations will be masked in the Dagster Cloud UI, and will provide a unique error ID that you can use to look up the error message in your own logs.
```

Comment:
Thoughts on adding a sentence that explains where the ID is located, using the screenshot as an example? It might not be obvious to folks using screen readers.

Reply:
Done!